codec/src/main/java/io/netty/handler/codec/DefaultHeaders.java

```diff
@@ -111,6 +144,7 @@ public DefaultHeaders(HashingStrategy<K> nameHashingStrategy,
         entries = new HeaderEntry[findNextPositivePowerOfTwo(max(2, min(arraySizeHint, 128)))];
         hashMask = (byte) (entries.length - 1);
         head = new HeaderEntry<K, V>();
+        this.valueValidator = valueValidator;
```

Comment:
Missing null check

Reply:
+1, consider verifying for null before any allocations